File: 1773_Count_Items_Matching_a_Rule/1773.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {       
        int ans = 0, idx;
        if (ruleKey == "type")
            idx = 0;
        else if (ruleKey == "color")
            idx = 1;
        else
            idx = 2;
        for (const auto item: items) {
            if (item[idx] == ruleValue)
                ans++;
        }
        return ans;
    }
};

int main() {
    Solution a;
    vector<vector<string>> v = {{"phone","blue","pixel"},{"computer","silver","lenovo"},{"phone","gold","iphone"}};
    string ruleKey = "color", ruleValue = "silver";
    cout << a.countMatches(v, ruleKey, ruleValue) << "\n";
    return 0;
}

File: 1773_Count_Items_Matching_a_Rule/1773.py
class Solution:
    def countMatches(self, items: list[list[str]], ruleKey: str, ruleValue: str) -> int:
        if ruleKey == 'type':
            idx = 0
        elif ruleKey == 'color':
            idx = 1
        else:
            idx = 2
        ans = 0
        for item in items:
            if item[idx] == ruleValue:
                ans += 1
        return ans

a = Solution()
print(a.countMatches(items = [["phone","blue","pixel"],["computer","silver","lenovo"],["phone","gold","iphone"]], ruleKey = "color", ruleValue = "silver"))
print(a.countMatches(items = [["phone","blue","pixel"],["computer","silver","phone"],["phone","gold","iphone"]], ruleKey = "type", ruleValue = "phone"))